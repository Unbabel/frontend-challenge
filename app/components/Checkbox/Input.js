import styled from 'styled-components';

const Input = styled.input`
  position: absolute;
  left: -99999px;
`;

export default Input;
