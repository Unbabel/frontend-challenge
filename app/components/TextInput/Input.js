import styled from 'styled-components';

const Input = styled.input`
  width: 100%;
  border: none;
  border-bottom: 2px solid transparent;
  outline: none;
  box-sizing: border-box;
  padding: 4px;
  color: ${props => props.theme.transcriptionTitleColor};
  font-weight: 600;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0;

  &:focus {
    border-bottom: ${props => `2px solid ${props.theme.blue}`};
  }
`;

export default Input;
