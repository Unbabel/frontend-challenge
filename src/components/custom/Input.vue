<template>
    <div contenteditable @input="updateValue">{{ modelValue }}</div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
    props: ['modelValue'],
    emits: ['update:modelValue'],
    methods: {
        updateValue(e: Event) {
            this.$emit('update:modelValue', (e.target as HTMLDivElement).innerText)
        }
    }
})
</script>

<style lang="scss" scoped>
div[contenteditable] {
    border: none;
    outline: none;
    font-size: 16px;
}
</style>