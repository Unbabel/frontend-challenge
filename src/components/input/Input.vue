<script>
	// Store
	import { mapState } from "vuex";

	export default {
		props: {
			value: {
				type: String,
			},
			placeholder: {
				type: String,
				default: "Insert text here",
			},
			customStyles: {
				type: String,
				default: "",
			},
			onChangeCallback: {
				type: Function,
			},
		},
		methods: {
			fitToContent() {
				this.$el.style.height = "5px";
				this.$el.style.height = this.$el.scrollHeight + "px";
			},
			onChangeHandler(ev) {
				this.fitToContent();
				this.onChangeCallback(ev.target.value);
			},
			getClasses() {
				const isColorBlindnessActive =
					this.$store.getters.isColorBlindnessActive;
				return `${this.$props.customStyles} ${
					isColorBlindnessActive && "colorBlindness"
				}`;
			},
		},
		mounted() {
			this.fitToContent();
		},
	};
</script>

<template>
	<textarea
		:class="getClasses()"
		:value="value"
		@input="onChangeHandler"
		:placeholder="placeholder"
	>
	</textarea>
</template>

<style scoped>
	textarea {
		width: 100%;
		border: 1px solid transparent;
		outline: none;
		resize: none;
	}

	textarea:focus {
		border: 1px solid var(--palette-lighter-grey);
		border-radius: 3px;
		background-color: var(--palette-dirty-white);
	}

	@media (max-width: 768px) {
		textarea {
			font-size: 0.9rem !important;
		}
	}
</style>
