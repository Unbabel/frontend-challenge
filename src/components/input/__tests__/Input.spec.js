import { describe, it, expect, beforeEach, vi } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import Input from "../Input.vue";

const mockCallback = vi.fn();
const mockFitToContent = vi.fn();
const mockGetClasses = vi.fn().mockImplementation(() => "colorBlindness");

describe("Input", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(Input, {
			propsData: {
				value: null,
				placeholder: "Insert text",
				onChangeCallback: mockCallback,
			},
			methods: {
				fitToContent: mockFitToContent,
				getClasses: mockGetClasses,
			},
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const textarea = wrapper.findAll("textarea");

		// Assert
		expect(textarea.exists()).toBe(true);
		expect(textarea).toHaveLength(1);
	});

	it("has correct classes", () => {
		// Assert
		expect(wrapper.classes()).toContain("colorBlindness");
	});
});
