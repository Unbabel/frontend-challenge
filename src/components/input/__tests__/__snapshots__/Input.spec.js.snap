// Vitest Snapshot v1

exports[`Input > matches snapshot 1`] = `"<textarea data-v-0f426a66=\\"\\" placeholder=\\"Insert text\\" class=\\"colorBlindness\\"></textarea>"`;
